graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGenerator.kt

```diff
@@ -90,14 +90,15 @@ open class SchemaGenerator(internal val config: SchemaGeneratorConfig) {
      * This function is recursive because while generating the additionalTypes it is possible to create additional types that need to be processed.
      */
     protected open fun generateAdditionalTypes(): Set<GraphQLType> {
-        val currentlyProcessedTypes = this.additionalTypes.toSet()
-        this.additionalTypes.clear()
-        val graphqlTypes = currentlyProcessedTypes.map { generateGraphQLType(this, it) }.toSet()
+        val currentlyProcessedTypes = mutableSetOf<KType>()
+        val graphqlTypes = mutableSetOf<GraphQLType>()
+        do {
+            currentlyProcessedTypes.addAll(this.additionalTypes)
+            this.additionalTypes.clear()
+            graphqlTypes.addAll(currentlyProcessedTypes.map { generateGraphQLType(this, it) })
+            currentlyProcessedTypes.clear()
```

Comment:
is constantly clearing the same object better than recreating the hash map?, e.g.

```
        val graphqlTypes = mutableSetOf<GraphQLType>()
        while (this.additionalTypes.isNotEmpty()) {
            val currentlyProcessedTypes = HashSet(this.additionalTypes)
            this.additionalTypes.clear()
            graphqlTypes.addAll(currentlyProcessedTypes.map { generateGraphQLType(this, it) })
        }
        return graphqlTypes.toSet()
```